ideascube/tests/test_utils.py

```diff
@@ -55,3 +55,30 @@ def test_size(tmpdir):
     with file_path.open("wb") as f:
         f.write(content)
     assert get_file_size(str(file_path)) == len(content)
+
+
+@pytest.mark.parametrize('string', [
+    'foo,bar,stuff,fuzzy',
+    'foo, bar, stuff, fuzzy',
+    'bar, stuff; fuzzy, foo',
+    ';;;; bar,   stuff    ;,fuzzy,foo  ',
+    'bar, Bar, stufF, FuZzY, fUzZY, FOO'
+])
+def test_tag_splitter_simple_tags(string):
+    oracle = set(['foo', 'bar', 'stuff', 'fuzzy'])
+    tags = tag_splitter(string)
+    assert len(tags) == len(oracle)
+    assert set(tags) == oracle
+
+
+@pytest.mark.parametrize('string', [
+    'black & white,stuff,fuzzy',
+    'black & white;fuzzy;stuff',
+    ';stuff; black & white,stuff,fuzzy',
+    '    black & white   ,,,, stuff  ,fuzzy',
+])
+def test_tag_splitter_tags_with_spaces(string):
+    oracle = set(['black & white', 'stuff', 'fuzzy'])
+    tags = tag_splitter(string)
+    assert len(tags) == len(oracle)
+    assert set(tags) == oracle
```

Comment:
Nice tests.

Great use of `parametrize`, so we can add more weird cases as we find (and fix) issues in the future.